### PR TITLE
Remove pandas dependency for CelebA dataset

### LINKED ISSUE #3650: Avoid pandas dependency for CelebA dataset
The CelebA dataset has a runtime dependency on `pandas`. Since all we need is `pandas.read_csv`, it would be interesting to see if we can replace `pandas.read_csv` by just the builtin `csv` module. The mergability of the PR would depend on how ugly / complex the code becomes... :)

cc @pmeier

### PR DESCRIPTION
**Pull request by carolineechen**
Remove the runtime dependency on `pandas` for CelebA, using Python's builtin `csv` module instead.

closes #3650

### CODE REVIEW

File: torchvision/datasets/celeba.py [hunk 1]
```diff
@@ -88,23 +92,44 @@ def __init__(
         }
         split_ = split_map[verify_str_arg(split.lower(), "split",
                                           ("train", "valid", "test", "all"))]
+        splits = self._load_csv("list_eval_partition.txt", header=None, index_col=0)
+        identity = self._load_csv("identity_CelebA.txt", header=None, index_col=0)
```
Comment (pmeier) [inline torchvision/datasets/celeba.py:96]: Given that `header=None` is the default value, I would simply drop it here. If you think it is important to pass it explicitly, I would remove the default value.
```suggestion
        splits = self._load_csv("list_eval_partition.txt", index_col=0)
        identity = self._load_csv("identity_CelebA.txt", index_col=0)
```

File: torchvision/datasets/celeba.py [hunk 2]
```diff
@@ -88,23 +92,44 @@ def __init__(
...
+        self.landmarks_align = landmarks_align.data[mask]
+        self.attr = attr.data[mask]
+        self.attr = (self.attr + 1) // 2  # map from {-1, 1} to {0, 1}
+        self.attr_names = attr.header
+
+    def _load_csv(
+        self,
+        filename: str,
+        header: int = None,
+        index_col: int = None
```
Comment (pmeier) [inline torchvision/datasets/celeba.py:115]: Given that all files here use `index_col=0`, I would simply hardcode the behavior and remove the parameter. No need to make this more general as it needs to be.

File: torchvision/datasets/celeba.py [hunk 3]
```diff
@@ -88,23 +92,44 @@ def __init__(
...
-        self.attr = torch.as_tensor(attr[mask].values)
-        self.attr = (self.attr + 1) // 2  # map from {-1, 1} to {0, 1}
-        self.attr_names = list(attr.columns)
+        with open(fn(filename)) as csv_file:
+            data = list(csv.reader(csv_file, delimiter=' ', skipinitialspace=True))
+
+        if header is not None:
+            headers = data[header]
+            data = data[header + 1:]
+        data_np = np.array(data)
```
Comment (pmeier) [inline torchvision/datasets/celeba.py:126]: Do we need the detour over `numpy` here? Can't we use `torch.tensor()` to begin with?

**Review by pmeier — COMMENTED** [review]
Hey @carolineechen, thanks a lot for the PR! I have a couple of minor comments below. Otherwise LGTM!

File: torchvision/datasets/celeba.py [hunk 4]
```diff
@@ -88,23 +91,42 @@ def __init__(
...
+        self.bbox = bbox.data[mask]
+        self.landmarks_align = landmarks_align.data[mask]
+        self.attr = attr.data[mask]
+        self.attr = (self.attr + 1) // 2  # map from {-1, 1} to {0, 1}
+        self.attr_names = attr.header
+
+    def _load_csv(
+        self,
+        filename: str,
+        header: int = None,
```
Comment (pmeier) [inline torchvision/datasets/celeba.py:113]: Nit: Make this explicit.

```suggestion
        header: Optional[int] = None,
```

**Review by pmeier — APPROVED** [review]
One minor nitpick left. Thanks a lot @carolineechen!

**Review by datumbox — APPROVED** [review]

**Pull request merged**